capabilities/writetarget/write_target_monitor.go

```diff
@@ -0,0 +1,221 @@
+package writetarget
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/protobuf/proto"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/beholder"
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+
+	"github.com/smartcontractkit/chainlink-framework/capabilities/writetarget/beholder/monitor"
+
+	"github.com/smartcontractkit/chainlink-framework/capabilities/writetarget/beholder/pb/data-feeds/on-chain/registry"
+	wt "github.com/smartcontractkit/chainlink-framework/capabilities/writetarget/beholder/pb/platform"
+	"github.com/smartcontractkit/chainlink-framework/capabilities/writetarget/beholder/pb/platform/on-chain/forwarder"
+)
+
+const (
+	repoCLLCommon = "https://raw.githubusercontent.com/smartcontractkit/chainlink-common"
+	// TODO: replace with main when merged
+	versionRefsDevelop = "refs/heads/generalized-monitoring-extraction"
+	schemaBasePath     = repoCLLCommon + "/" + versionRefsDevelop + "/pkg/capabilities/writetarget/pb"
+)
+
+// NewWriteTargetMonitor initializes a Beholder client for the Write Target
+//
+// The client is initialized as a BeholderClient extension with a custom ProtoEmitter.
+// The ProtoEmitter is proxied with additional processing for emitted messages. This processing
+// includes decoding messages as specific types and deriving metrics based on the decoded messages.
+// TODO: Report decoding uses the same ABI for EVM and Aptos, however, future chains may need a different
+// decoding scheme. Generalize this in the future to support different chains and decoding schemes.
+func NewWriteTargetMonitor(ctx context.Context, lggr logger.Logger) (*monitor.BeholderClient, error) {
```

Comment:
We are already in `package writetarget`. No need to repeat:
```suggestion
func NewMonitor(ctx context.Context, lggr logger.Logger) (*monitor.BeholderClient, error) {
```